mux.go

```diff
@@ -51,6 +51,9 @@ type Router struct {
 	// Configurable Handler to be used when the request method does not match the route.
 	MethodNotAllowedHandler http.Handler
 
+	// If true, only handle MethodNotAllowed if the router MethodNotAllowedHandler is explicitly specified
+	RequireExplicitMethodNotAllowedHandler bool
```

Comment:
We can likely carry private state here as we know when the user has set the handler [above[